qiskit/chemistry/transformations/bosonic_transformation.py

```diff
@@ -264,13 +264,13 @@ def _build_single_hopping_operator(index: List[List[int]],
 
         dummpy_op = BosonicOperator(np.asarray(hml, dtype=object), basis)
         qubit_op = dummpy_op.mapping(qubit_mapping)
-        if len(qubit_op.paulis) == 0:
+        if qubit_op.is_zero():
             qubit_op = None
 
         return qubit_op
 
     def build_hopping_operators(self, excitations: Union[str, List[List[int]]] = 'sd') \
-            -> Tuple[Dict[str, WeightedPauliOperator],
```

Comment:
I am guessing that deprecating the use of the existing type, while adding PauliSumOp, for compatibility over the transition before its removed later is problematic overall given some of the methods return this a type. I will note that the legacy operators were not deprecated as such before, since they still had to be used due to lack of Z2Symmetries.

Reply:
WeightedPauliOperator will be deprecated before this PR Qiskit/qiskit-terra#5537.

Reply:
Yes, but if anyone called build_hopping_operators method the existing WeightedPauliOperator, now deprecated, is not supported though still. Arguably it should still be supported, but deprecated, along with the new replacement as a Union of types. As I indicated I imagine this might be problematic to do this and it may have to stay a possible breaking change on the APIs if anyone is using any of these. (I am sure we can adjust all Aqua etc to compensate - hopefully no tutorials use any of these method and I hope users in general do no either, though since they are public interface they might)